eo-parser/src/main/java/org/eolang/parser/errors/ParsingErrors.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2024 Objectionary.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.parser.errors;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import org.antlr.v4.runtime.BaseErrorListener;
+import org.antlr.v4.runtime.RecognitionException;
+import org.antlr.v4.runtime.Recognizer;
+import org.cactoos.Text;
+import org.cactoos.list.ListOf;
+import org.eolang.parser.ParsingException;
+import org.xembly.Directive;
+
+/**
+ * Accumulates all parsing errors.
+ *
+ * @since 0.30.0
+ */
+public final class ParsingErrors extends BaseErrorListener implements Iterable<Directive> {
```

Comment:
@volodya-lombrozo same here: if it's an iterable of `Directive`, the `Dr` prefix will be a good name pattern, I believe

Reply:
@volodya-lombrozo also, implementation inheritance is not what we like to use. Composition is a better choice usually. See https://www.yegor256.com/2016/09/13/inheritance-is-procedural.html

Reply:
@yegor256 Renamed to `DrParsingErrors`.

Reply:
> @volodya-lombrozo also, implementation inheritance is not what we like to use. Composition is a better choice usually. See https://www.yegor256.com/2016/09/13/inheritance-is-procedural.html

I would like to avoid inheritance, but `BaseErrorListener` is an ANTLR class and it's the way how they handle errors.